sdk/src/azure/iot/az_iot_provisioning_client.c

```diff
@@ -276,6 +277,35 @@ AZ_INLINE az_result _az_iot_provisioning_client_parse_payload_error_code(
   return AZ_ERROR_ITEM_NOT_FOUND;
 }
 
+AZ_INLINE az_result
+_az_iot_provisioning_client_get_json_object_span(az_json_reader* jr, az_span* out_span)
+{
+  *out_span = jr->token.slice;
+
+  if (jr->token.kind == AZ_JSON_TOKEN_NULL)
+  {
+    *out_span = AZ_SPAN_EMPTY;
+    return AZ_OK;
```

Comment:
Interesting, why set `out_span` as empty, rather than what it is set to, above (on line 283), which is the characters from the JSON payload, i.e. `null`?

Reply:
The purpose of this function is to extract a complex JSON object that can be parsed by the application.
`null` isn't a valid Json and so we return an empty span.